src/matroskavideo.cpp

```diff
@@ -655,7 +655,7 @@ namespace Exiv2 {
             switch (mt->val_) {
             case 0x0489:
                 if(size <= 4) {
-                    duration_in_ms = static_cast<int64_t>(getFloat(buf, bigEndian) * time_code_scale * 1000);
+                    duration_in_ms = static_cast<int64_t>(getFloat(buf, bigEndian) * static_cast<float>(time_code_scale * 1000));
```

Comment:
[suggestion] I would replace `1000` to `1000.0` to avoid the implicit conversion

Reply:
We get the implicit conversion anyway, since `time_code_scale` is a double. I have now cast `time_scale_code` itself to a float, that way we only have to multiply two floats and not two doubles.